### ISSUE TITLE
2.1.0 compile error

### ISSUE THREAD

**Issue by navied** [sass/sassc-ruby#141]
2.1.0 doesn't seem to be compile for me on alpine linux systems. It seems to be failing with this error:

LoadError: Could not open library '/codebuild/output/src471049011/src/vendor/bundle/ruby/2.5.0/gems/sassc-2.1.0-x86_64-linux/lib/sassc/libsass.so': Error loading shared library ld-linux-x86-64.so.2: No such file or directory (needed by /codebuild/output/src471049011/src/vendor/bundle/ruby/2.5.0/gems/sassc-2.1.0-x86_64-linux/lib/sassc/libsass.so) 

It seems to compile fine on a Osx / ubuntu system just fine though!

**Comment by FireFart** [comment]
Getting the same error after upgrading to rails 6

**Comment by glebm** [comment]
On Alpine you may need to install the `libc6-compat` package.

**Comment by glebm** [comment]
Note also that this is not a compilation error: it's an error while loading the pre-compiled library. Alpine comes with `musl` instead of `libc6` which the pre-compiled `libsass.so` is linked against. If you install the gem from source on Alpine (`gem install sassc --platform RUBY`), `libsass.so` will be linked against `musl`.

**Comment by FireFart** [comment]
Sorry buy installing `libc6-compat` is no real option.

I am currently installing the gem from source with bundler (`bundle install --system --clean --no-cache`) in the `ruby-aline` image and it still uses the compiled version so I clearly consider this as a bug as it should compile the library during install

**Comment by glebm** [comment]
> I am currently installing the gem from source 

No, the following line means you're installing the pre-compiled version:

> Installing sassc 2.1.0 (**x86_64-linux**)

Here is how you can force bundler to avoid pre-compiled gems:

```
bundle config force_ruby_platform true
```

(This has the same effect as setting `BUNDLE_FORCE_RUBY_PLATFORM=true` env var).

Run this in your Alpine container before running `bundle install`.

**Comment by FireFart** [comment]
Thanks the bundle config worked!
Maybe this should be mentioned in the Readme too so other people can spot this easiliy?

**Comment by glebm** [comment]
By the way, this is something you should always do if you're using Alpine. It's not specific to this gem at all.

How did you decide to use Alpine? It's designed for power users (https://alpinelinux.org/about/) who understand all the implications of using a non-libc environment. There are small Debian containers as well, such as Minideb, though still much larger than Alpine.

**Comment by FireFart** [comment]
It's not about not understanding Alpine linux, it's about a not properly documented flag in bundler. Nothing in the docs hints at this and honestly I've never seen it before.
Take the `nokogiri` and `ffi` gems for example. Both build on install with native extensions without using this flag, but this gem doesn't. We've ran the images for years now without problems but the recent updates from rails introduced these problems (maybe they were always there but the libraries were linked statical).
But all after all this now works

**Comment by glebm** [comment]
Nokogiri and ffi don't have pre-compiled gems for Linux, that's why they are always compiled from source on install on Linux.

Rails update has nothing to do with it, except depending on sassc v2.1.0+ under the hood by default.

glibc should not be linked statically (and never is).

**Comment by alexwlchan** [comment]
Just to note:

I ran into this problem when I upgraded to Jekyll 4, which I run in a Docker container with an Alpine base image.

Adding `BUNDLE_FORCE_RUBY_PLATFORM=true` to my Docker image seems to have done the trick, but I got a segfault first (https://github.com/alexwlchan/alexwlchan.net/issues/286), and it took a while to realise that Alpine was the culprit.

**Comment by L1Cafe** [comment]
> @glebm Should we perhaps stop shipping a precompiled gem for linux? We've got a few different issues at this point, and I'm not sure how to debug them. These issues didn't really come up previously.

I think my issue #144 is also related to this one. The Jekyll Docker image is indeed based off Alpine: https://github.com/envygeeks/jekyll-docker#packages. Will try the `bundle config` trick now

**Comment by glebm** [comment]
> Should we perhaps stop shipping a precompiled gem for linux?

@bolandrm Yes, I think we should. It's unfortunate that RubyGems doesn't let us to target a particular version of libc and users don't seem to be aware of this distinction because other popular gems don't ship precompiled Linux versions.

**Comment by L1Cafe** [comment]
Thanks @glebm! It now works on my Alpine Docker machine without the `bundle config` setting. Thanks for the fast response.
